### PR TITLE
CASEFLOW-106 #done Allow CAVC Litigation Support Users to Correct Issues

### LINKED ISSUE #15289: Allow CAVC Litigation Support Users to Add Issues on newly created appeal
## User or job story
User story: As a CAVC Litigation Support user, I need the ability to add issues on a CAVC case that the Board may not have captured in the original decision, so that the new issue can be decided on. 

## Acceptance criteria
- [ ] This feature should be accessible to the following user groups: CAVC Litigation Support
- [ ] Link should only appear if there is an active `CavcTask`
- [ ] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: CAVC Remand
- [ ] User is routed to the Edit Issues page in Caseflow intake
- [ ] VBMS records are created as needed

### PR DESCRIPTION
**Pull request by yoomlam**
Resolves #15289 aka [CASEFLOW-106](https://vajira.max.gov/browse/CASEFLOW-106)

### Description
As a CAVC Litigation Support user, I need the ability to add issues on a CAVC case that the Board may not have captured in the original decision, so that the new issue can be decided on.

### Acceptance Criteria
- [ ] Verify the user has the ability to add issues on the front end after appeal is created
- [ ] Verify the link only appears to CAVC Litigation Support user if there is an active CavcTask
- [ ] Verify the user is routed to the Edit Issues page in Caseflow intake
- [ ] ~Verify VBMS records are created as needed~ (UPDATE: Sally says, "Since appeals are processed in Caseflow, nothing is updated in VBMS when someone adds a new issue to an appeal. A VBMS update happens for Comp/Pension/Fiduciary HLRs and SCs" -- [Slack](https://dsva.slack.com/archives/C54QCEHAL/p1606323959368000?thread_ts=1606319918.367400&cid=C54QCEHAL).

### Testing Plan
1. Log in as CAVC_LIT_SUPPORT_ADMIN (VACO)
1. Go to http://localhost:3000/organizations/cavc-lit-support?tab=unassignedTab&page=1
1. Assign a case to yourself
1. Open the case details page for the case (under the Assigned tab of the Team view -- same URL above)
1. Click the "Correct issues" link 
1. Add issue(s) via the "Add issue" button. In the modal, click "None of these match, see more options" to add an entirely new issue (that is eligible). Optionally, remove issue(s).
1. Click "Save" and confirm.
1. Back in the case details page, eligible issues are shown, and removed or ineligible issues are not shown.
1. Bonus: Check associated `RequestIssuesUpdate` records for appropriate modifications.
1. Bonus: Repeat the steps above. May have to `riu.perform!` if you get an error message after trying to save subsequent edits. See [VBMS investigation comment below](https://github.com/department-of-veterans-affairs/caseflow/pull/15631#issuecomment-733774776).

### User Facing Changes
 - [x] Screenshots of UI changes added 

New "Correct Issues" link for CAVC Litigation Support users
image

After adding 2 new issues:
image

Clicking "Save" brings up confirmation:
image

After saving, case details shows success banner (and new issues if they were eligible):
image

Removing an issue shows confirmation modal:
image

Removing last issue shows case-will-be-closed warning modal:
image

### CODE REVIEW

File: app/policies/appeal_request_issues_policy.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@ def case_is_not_in_active_review?
     Task.where(appeal: appeal, type: "AttorneyTask").empty?
   end
 
-  def hearing_is_assigned_to_judge_user
+  def hearing_is_assigned_to_judge_user?
```
Comment (yoomlam) [inline app/policies/appeal_request_issues_policy.rb:30]: For consistency with other methods that return a boolean.

File: client/app/intake/pages/addIssues.jsx [hunk 2]
```diff
@@ -137,7 +137,9 @@ class AddIssuesPage extends React.Component {
   establishmentCredits() {
     return <div className="cf-intake-establish-credits">
       Established {this.establishmentCreditsTimestamp()}
-      <span> by <a href={`/intake/manager?user_css_id=${this.props.intakeUser}`}>{this.props.intakeUser}</a></span>
+      { this.props.intakeUser &&
+        <span> by <a href={`/intake/manager?user_css_id=${this.props.intakeUser}`}>{this.props.intakeUser}</a></span>
+      }
```
Comment (yoomlam) [inline client/app/intake/pages/addIssues.jsx:142]: To address [rspec failures](https://app.circleci.com/pipelines/github/department-of-veterans-affairs/caseflow/36852/workflows/1fc5f683-7491-4bd3-a36b-7efa3773d82e/jobs/137581/parallel-runs/10?filterBy=FAILED) due to `Sniffybara::PageNotAccessibleError: Links must have discernible text`, need to hide `intakeUser` link if it is not set.

File: spec/feature/queue/cavc_task_queue_spec.rb [hunk 3]
```diff
@@ -10,6 +10,59 @@
   let!(:org_nonadmin2) { create(:user, full_name: "Tooey Remandy") { |u| CavcLitigationSupport.singleton.add_user(u) } }
   let!(:other_user) { create(:user, full_name: "Othery Usery") }
 
+  include IntakeHelpers
```
Comment (yoomlam) [inline spec/feature/queue/cavc_task_queue_spec.rb:13]: Reusing `IntakeHelpers` from https://github.com/department-of-veterans-affairs/caseflow/blob/d924e2e167ccdc93eb7a26ba1e02d9ec02ee8a76/spec/feature/queue/delete_request_issues_spec.rb#L4

**Comment by yoomlam** [comment]
Attempting to confirm writes to VBMS (after issue edits are saved):
- a RequestIssuesUpdate (RIU) is created: https://github.com/department-of-veterans-affairs/caseflow/blob/15de1d3e26cff73a716e308d0e10d6df09d88e9e/app/controllers/appeals_controller.rb#L167-L173
- http://localhost:3000/asyncable_jobs/RequestIssuesUpdate/jobs shows the RIU
- these return correct values:
```ruby
riu.added_issues
riu.removed_issues
riu.establish! # results in the RIU being removed from the asyncable_jobs page
```
- new issues are created by `process_issues!` when the `riu.perform!` is called (by the asyncable_job processor). This will also trigger a `DecisionReviewProcessJob` to eventually run. https://github.com/department-of-veterans-affairs/caseflow/blob/8f373137c017f72f43b7505fa6615644bf8a6ebe/app/models/request_issues_update.rb#L154-L161
   - when a new issue is created in `appeal.create_issues!(added_issues)`, VACOLS issues are created for legacy appeals https://github.com/department-of-veterans-affairs/caseflow/blob/6bf7df400ed692b8d1f064b13bb99a80c29937a0/app/models/request_issue.rb#L609-L612
- VBMS is called in `establish!` (which is called async via `DecisionReviewProcessJob`): https://github.com/department-of-veterans-affairs/caseflow/blob/8f373137c017f72f43b7505fa6615644bf8a6ebe/app/models/request_issues_update.rb#L53-L71
   - updates `user` on EPEs (if the appeal was processed in VBMS)
   - for each removed issue that has an EPE, `RequestIssueContention.new(request_issue).remove!` (which calls `VBMSService.remove_contention!`) and then call `cancel_unused_end_product!` on each EPE.

**Comment by yoomlam** [comment]
Jira ticket shows this PR and commit messages containing "CASEFLOW-106":
image

**Review by hschallhorn — COMMENTED** [review]
```ruby
review = SendCavcRemandProcessedLetterTask.open.last.appeal
review.uuid
```
Add some issues, all looks gravy! (you can add eligible issues by selecting "none of these match" and creating a new issue if you want to see case details update with newly added issues)

needed to actually process the request issues before editing them again
```ruby
RequestIssuesUpdate.processable.find_by(review: review)&.establish!
```

Removed all eligible issues, which cancelled the appeal. @yoomlam did we confirm with product or design that cavc users should be able to cancel an entire appeal?

File: app/policies/appeal_request_issues_policy.rb [hunk 4]
```diff
@@ -38,4 +38,9 @@ def case_is_in_active_review_by_current_user?
       status: [Constants.TASK_STATUSES.assigned, Constants.TASK_STATUSES.in_progress]
     ).select { |task| task.is_a?(JudgeTask) || task.is_a?(AttorneyTask) }.any?
   end
+
+  def cavc_task_open?
```
Comment (hschallhorn) [inline app/policies/appeal_request_issues_policy.rb:42]: We're checking for more than just an open task here. Can we name this something a bit more descriptive? maybe `editable_by_cavc_team_member?`

File: spec/feature/queue/cavc_task_queue_spec.rb [hunk 5]
```diff
@@ -10,6 +10,59 @@
...
+        expect(appeal.request_issues.count).to eq 1
+        click_on "+ Add issue"
+        add_intake_nonrating_issue(
+          category: "Unknown issue category",
+          description: new_issue_description,
+          date: (Time.zone.now - 100.days).mdY
+        )
+        click_edit_submit_and_confirm
+        expect(page).to have_content "Edit Completed"
+        expect(page).to have_content "You have successfully added 1 issue."
```
Comment (hschallhorn) [inline spec/feature/queue/cavc_task_queue_spec.rb:51]: Let's add a check that the new issue shows up on case details without a refresh (can probably just check that the page has the `new_issue_description`). It does in manual testing 🎉

**Review by hschallhorn — APPROVED** [review]
⏩  Fabuloso! Digging this minimal additions to implement features! Also A+ work tackling the atty issue!

**Comment by yoomlam** [comment]
Woohoo! Jira updated automatically: 
image

**Pull request merged**